### ISSUE TITLE
Pyright v1.1.370 has incorrect min vscode engine version as 1.86

### ISSUE THREAD

**Issue by aadisriram** [microsoft/pyright#8344]
**Describe the bug**
Pyright v1.370+ has the engine version in package.json set to 1.86, however when starting pyright on 1.86 the following error is thrown in the console:

```
The language client requires VS Code version ^1.89.0 but received version 1.86.<>
```

**Code or Screenshots**
This is happening because the languageserver node package pyright depends on has a min engine version of 1.89
https://github.com/microsoft/vscode-languageserver-node/blob/release/client/10.0.0-next.6/client/package.json

<img width="1382" alt="image" src="https://github.com/microsoft/pyright/assets/792771/bc929a4a-1f61-4b03-93af-8e14b8748e04">

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

@debonte, @rchiodo, looks like this was broken by a recent change that came in through a pylance update.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.371.
